# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1425428716 21600
#      Tue Mar 03 18:25:16 2015 -0600
# Node ID 2eed41ae3b6474644dc5f793d83a215459c0a634
# Parent  371aa9261b1a87a8760262480e8fbb34f5aff69a
Fix computation of next-power-of-2 for height in OpenGL texture creation.

* gl-render.cc (opengl_texture::create): Use 'h', not 'w', as argument to
compute 'th'.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -173,17 +173,17 @@ opengl_texture::create (const octave_val
       // FIXME: dim_vectors hold octave_idx_type values.
       //        Should we check for dimensions larger than intmax?
       int h, w, tw, th;
       h = dv(0), w = dv(1);
       GLuint id;
       bool ok = true;
 
       tw = next_power_of_2 (w);
-      th = next_power_of_2 (w);
+      th = next_power_of_2 (h);
 
       glGenTextures (1, &id);
       glBindTexture (GL_TEXTURE_2D, id);
 
       if (data.is_double_type ())
         {
           const NDArray xdata = data.array_value ();
 
